addons/mail/models/mail_canned_response.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class MailCannedResponse(models.Model):
+    """
+    Canned Response: content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.
+    """
+
+    _name = "mail.canned_response"
+    _description = "Canned Response"
+    source = fields.Char(
+        "Shortcut",
+        required=True,
+        index="trigram",
+        help="Canned response that will automatically be substituted with longer content in your messages."
+        " Type ':' followed by the name of your shortcut (e.g. :hello) to use in your messages.",
+    )
+    substitution = fields.Text(
+        "Substitution",
+        required=True,
+        help="Content that will automatically replace the shortcut of your choosing. This content can still be adapted before sending your message.",
+    )
+    description = fields.Char("Description")
+    last_used = fields.Datetime("Last Used", help="Last time this canned_response was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="mail.canned_response_administrator")
+    is_shared = fields.Boolean(
+        string="Determines if the canned_response is currently shared with other users",
+        compute="_compute_is_shared",
+        store=True,
+    )
+    is_editable = fields.Boolean(
+        string="Determines if the canned response can be edited by the current user", compute="_compute_is_editable"
+    )
+
+    @api.depends("create_uid")
+    def _compute_is_editable(self):
+        for canned_response in self:
+            canned_response.is_editable = bool(
```

Comment:
It is strange, as it does not match ACLs. Why do you need this field anyway ? Can't you do a ``check_access_rules('write')``instead if you need to know it ? Usage in views is useless, should be done with classic ACLs, it avoids to define several times the same thing and is more standard.

Reply:
ACL's don't work just on fields. And don't set readonly attribute in view. This is why I compute this field.
But I can still use `check_access_rule` in the compute.